src/player.c

```diff
@@ -1304,6 +1315,44 @@ void player_targetPlanetSet( int id )
 }
 
 
+/**
+ * @brief Sets the player's target asteroid.
+ *
+ *    @param id Target planet or -1 if none should be selected.
+ */
+void player_targetAsteroidSet( int field, int id )
+{
+   int old, i;
+   AsteroidAnchor *anchor;
+   Asteroid *ast;
+   AsteroidType *at;
+   Commodity *com;
+
+   if ((player.p == NULL) || pilot_isFlag( player.p, PILOT_LANDING ))
+      return;
+
+   old = player.p->nav_asteroid;
+   player.p->nav_asteroid = id;
+   if (old != id) {
+      if (id >= 0)
+         player_soundPlayGUI(snd_nav, 1);
+
+         /* Recover and display some info about the asteroid */
+         anchor = &cur_system->asteroids[field];
+         ast = &anchor->asteroids[id];
+         at = space_getType( ast->type );
+
+         player_message( _("Asteroid targeted, composition: ") );
```

Comment:
Adding this was a nice touch. But perhaps the player should have to buy a special scanner to get this information?